futures-02.md

```diff
@@ -0,0 +1,396 @@
+
+### `futures-io`
+
+`futures-io` includes the updated `AsyncRead` and `AsyncWrite` traits.
+These traits were previously included in the `tokio-io` crate.
+However, `AsyncRead` and `AsyncWrite` don't have any dependencies on the `tokio`
+crate or the rest of the `tokio` stack, so they will be moved into the `futures`
+group of libraries. `futures-io` depends only on the `futures-core` crate
+and the `bytes` crate (for the [`Buf`] and [`BufMut`] traits).
+
+[`Buf`]: https://carllerche.github.io/bytes/bytes/trait.Buf.html
+[`BufMut]: https://carllerche.github.io/bytes/bytes/trait.BufMut.html
+
+### `futures-sink`
+
+`futures-sink` includes the `Sink` trait, which allows values to be sent
+asynchronously. The `Sink` trait is excluded from `futures-core` because its
+design is much less stable than `Future` and `Stream`. There are several
+potential breaking changes to the `Sink` trait under consideration.
+
+This crate will depend only on `futures-core`.
+
+### `futures-channel`
+
+`futures-channel` includes the `mpsc` and `oneshot` messaging primitives.
+This crate is a minimal addition on `futures-core` designed for low churn
+so that channels may appear in the public APIs of libraries.
+
+### `futures-util`
+
+`futures-util` contains a number of convenient combinator functions for
+`Future`, `Stream`, and `Sink`, including the `FutureExt`, `StreamExt`,
+and `SinkExt` extension traits.
+
+This crate depends on `futures-core`, `futures-io`, and `future-sink`.
+
+### `futures-executor`
+
+`futures-executor` contains the `Executor` trait which allows spawning
+asynchronously tasks onto a task executor. Like `Sink`, this is a less
+mature API which hasn't seen as much practical use. There are several
+possible breaking changes to this API coming in the near future.
+
+This crate depends on `futures-core`.
+
+## Small, Breaking API Improvements and Clarifications
+
+- `Task` will be renamed to `Waker`. This name more clearly reflects how
+  the type is used. It is a handle used to awaken tasks-- it is not a `Task`
+  itself.
+- `Async` variant `NotReady` will be renamed to `Pending`.
+- `AsyncRead` and `AsyncWrite` no longer have `io::Read` and `io::Write`
+  bounds. Discussion on this can be found [in this Github issue.](https://github.com/tokio-rs/tokio-io/issues/83)
+- `AsyncRead::prepare_uninitialized_buffer` has been replaced with an
+  `initializer` function similar to [the unstable one on `io::Read`.](https://doc.rust-lang.org/std/io/trait.Read.html#method.initializer)
+  This resolves the issue of the `unsafe` on `prepare_uninitialized_buffer`
+  being `unsafe` to implement, *not* `unsafe` to use
+  (the more common, and arguably the correct meaning of `unsafe fn`).
+- `AsyncRead` and `AsyncWrite` now offer separate methods for
+  non-vectorized and vectorized IO operations. This makes the API easier to
+  understand, makes the traits object-safe, and removes the dependency on
+  the `bytes` crate.
```

Comment:
But adds one on `io-vec`, right?

Reply:
The dependency on `IoVec` exists either way since it's exposed through the interface of `bytes`.